### PR TITLE
Apodization for frequency monitors

### PR DESCRIPTION
**Pull request by dbochkov-flexcompute**
Specification of apodization for frequency monitors

note that I set `width` to be a non-optional parameters, because generally there doesn't seem to be a common default value

### CODE REVIEW

File: tidy3d/components/apodization.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    end: pd.NonNegativeFloat = pd.Field(
+        inf,
+        title="End Interval",
+        description="Defines the time at which the end apodization begins.",
+        units=SECOND,
+    )
+
+    width: pd.PositiveFloat = pd.Field(
+        ...,
```
[the remark this answers is not in the document]
Comment (dbochkov-flexcompute) [reply tidy3d/components/apodization.py:36]: I like it, however I am worried that a user might not realize/just forget they have to set `width`, so they would use `ApodizationSpec(start=1, end=2)` and not get any apodization in their results. I guess we could set all `start`, `end`, and `width` to `None` by default, and a validator that checks if either `start` or `end` is not `None`, then `width` must not be `None` as well.

[the remark this answers is not in the document]
Comment (momchil-flex) [reply tidy3d/components/apodization.py:36]: This makes sense to me yeah.

File: tidy3d/components/apodization.py [hunk 2]
```diff
@@ -4,49 +4,58 @@
...
+        """Ensure end is greater than or equal to start."""
         start = values.get("start")
-        if val and val < start:
+        if val is not None and start is not None and val < start:
             raise SetupError("End apodization begins before start apodization ends.")
         return val
 
+    @pd.validator("width", always=True, allow_reuse=True)
+    def width_provided(cls, val, values):
+        """Ensure end is greater than or equal to start."""
```
Comment (momchil-flex) [inline tidy3d/components/apodization.py:52]: Wrong docstring

**Review by momchil-flex — APPROVED** [review]

File: tidy3d/components/apodization.py (hunk 2 above)
Comment (dbochkov-flexcompute) [reply tidy3d/components/apodization.py]: My bad, fixed!

**Pull request merged**